### DISCUSSION TITLE
Initial Design for Report Creation

### DISCUSSION THREAD

**Discussion by JorgeRivera94** [uprm-inso4116-2024-2025-s1/semester-project-SafeRUM#171] [Ideas]
Below is initial intended user flow for creating a report once the user leaves the landing page to create a report.

![Screenshot 2024-09-20 072251](https://github.com/user-attachments/assets/4e88b290-3fd9-4dca-ab9a-cd373844d365)

Category Selection Page:
* Button space should take up most of the screen to make it easy for the reporter to submit their report.
* Should be minimal to provide as much space as possible for the options to stand out.
* The selected option will serve as a category tag for the created report.
* Selecting "Safety Issue", "Suspicious Person", "Armed Suspect" or "Sexual Harasser" will take the user to the Report Details page.
* Selecting "SOS" will take the user to the SOS Options page.
* Selecting the back arrow button will take the user the app's landing page.

Report Details Page:
* Contains a short text input for the report title
* Contains a large text input area for the report description
* Contains a Detected Location widget similar to that in Jotform. This widget must be supported by a location API like Google Maps Platform API.
* Contains a button to submit.
* Each area must make use of its space to facilitate user input.
* Selecting the back arrow button will take the user the Category Selection page.
* Selecting the "Submit" button will take the user to the app's landing page.

SOS Options Page:
* Contains a Detected Location widget similar to that in Jotform. This widget would be supported by a location API like Google Maps Platform API.
* Selecting the "Panic Alert" button will take the user to the app's landing page.
* Selecting the "Immediate Help" button will take the user to the app's landing page.
* Selecting the back arrow button will take the user to the Category Selection page.
* "Panic Alert" and "Immediate Help" buttons must take up most of the screen for faster alert submission.

Due to lack of experience, button sizes are not as big as intended in the above descriptions.

Below are the frames in presentation mode for a small Android device.

Category Selection Page
![Screenshot 2024-09-20 020553](https://github.com/user-attachments/assets/129cbab2-c4de-444c-8369-28a94110db07)

Report Details Page:
![Screenshot 2024-09-20 020608](https://github.com/user-attachments/assets/fce48e81-2bef-4c4e-a511-6cdf04624577)

SOS Options Page:
![Screenshot 2024-09-20 020627](https://github.com/user-attachments/assets/2f8d857d-f3bc-48be-b61e-d216fbecde88)

Feedback will be appreciated. Specially in button colors. I tried for to make the "Submit" button to be a neutral color, as Green would be similar to the background and warmer colors would give a sense of pause. The orange of the non "SOS" buttons are Orange as maintaining the color as white may reduce the weight of these situations.

**Comment by YanehRuiz** [comment]
Hi there! I really love the color palette you've chosen. However, to ensure it aligns better with the rest of the designs and to enhance visibility, I’d suggest muting the colors a bit for improved readability.

**Reply by JorgeRivera94** [reply]
Thank you! Will be worked on starting Milestone 2.

**Reply by JorgeRivera94** [reply]
Hi! Link posted on issue #176 for feedback.

**Reply by JorgeRivera94** [reply]
Manager approved changes updated in documentation.
https://docs.google.com/document/d/1_wP3GOpOTexKhcp-WD7vQ7LvWcJn5lpBgYzgaBjY_AE/edit?usp=sharing
